src/GovUk.Education.ExploreEducationStatistics.Common/Services/BlobStorageService.cs

```diff
@@ -37,7 +37,7 @@ namespace GovUk.Education.ExploreEducationStatistics.Common.Services
     /// changes that make it difficult to upgrade when relying on lower-level
     /// details e.g. Azure SDK v12 is an entirely new package compared to v11.
     /// </summary>
-    public class BlobStorageService : IBlobStorageService
+    public class BlobStorageService : IPrivateBlobStorageService, IPublicBlobStorageService
```

Comment:
Aha interesting!  I would've assumed there'd be concrete PrivateBlobStorageService and PublicBlobStorageService classes that extended an abstract `BaseBlobStorageService` but this is a nice way of doing this with less classes and boilerplate :)

Reply:
Although nearer the end of my review, I've done a switcheroo and suggested that maybe having concrete classes would actually be more beneficial and lead to simpler code in `Startup.cs` where you're having to construct these manually each time.

Reply:
I originally did have classes, but for some reason decided against it. I have now restored the classes and it all seems fine. I ran the UI tests and got no failures.

Reply:
One thing to note, to achieve this I've had to add an empty constructor to `BlobStorageService` and change `BlobStorageService` variables to `protected` with no `readonly`. But I think this is fine.

Reply:
Yup sounds good to me!

Reply:
I'd also make this class abstract now so no one can accidentally construct one by accident.